scripts/detect_errors.py

```diff
@@ -0,0 +1,15 @@
+
+import sys
+
+try:
+    for line in sys.stdin:
+        sys.stdout.write(line)
+        if 'error' in line: raise sys.exit()
```

Comment:
This `raise sys.exit()` is confusing. `sys.exit()` already works by raising an exception, so it's like writing `raise (raise SystemExit)` (if that was valid syntax, which it's not).

I think you actually don't need any `try`/`except` to do this, just delete all those lines and the `raise sys.exit()` and fix the block structure.